lms/static/scripts/frontend_apps/utils/vitalsource.js

```diff
@@ -0,0 +1,14 @@
+/**
+ * A naive regex matcher for a VBID in a URL-like string.
+ *
+ * @param {string} url
+ * @returns {string|null} VBID, or `null` if nothing looks like a VBID in `url`
+ */
+export function bookIDFromURL(url) {
+  const bookIdPattern = /book\/([0-9A-Z-]+)(\/|$)/;
```

Comment:
Would be useful to have a comment about where the charset comes from. IIRC we discussed this on Slack somewhere, but I think it would be useful to include a brief summary here.

Reply:
I have considerably expanded the function doc comment.